nemo/collections/tts/models/ed_mel2spec.py

```diff
@@ -0,0 +1,335 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Sequence
+
+import numpy as np
+import soundfile as sf
+import torch
+import torch.nn.functional as F
+from hydra.utils import instantiate
+from omegaconf import MISSING, DictConfig, OmegaConf, open_dict
+from torch import Tensor, nn
+
+from nemo.collections.tts.helpers.helpers import eval_tts_scores, griffin_lim
+from nemo.collections.tts.models.base import MelToSpec
+from nemo.collections.tts.modules.ed_mel2spec import OperationMode
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+from nemo.core.neural_types.elements import MelSpectrogramType, SpectrogramType
+from nemo.core.neural_types.neural_type import NeuralType
+from nemo.utils import logging
+
+
+@dataclass
+class EDMel2SpecConfig:
+    mel2spec: Dict[Any, Any] = MISSING
+    train_ds: Optional[Dict[Any, Any]] = None
+    validation_ds: Optional[Dict[Any, Any]] = None
+    train_params: Optional[Dict[Any, Any]] = None
+    sched: Optional[Dict[Any, Any]] = None
+
+
+def gen_filter(k):
+    K = torch.ones(1, 1, k, 1)
+    K.requires_grad = False
+    return K
+
+
+class EDMel2SpecModel(MelToSpec):
+    """A model that convert mel spectrograms to linear spectrograms"""
```

Comment:
Why do you call this an EncoderDecoder model? Isn't it just a stack of convolutions?